xls/examples/parameterized/BUILD

```diff
@@ -0,0 +1,99 @@
+load("//xls/build_rules:xls_utilities.bzl", "proto_to_dslx")
+load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
+load(
+    "//xls/build_rules:xls_build_defs.bzl",
+    "xls_dslx_ir",
+    "xls_dslx_library",
+    "xls_ir_opt_ir",
+    "xls_ir_verilog",
+)
+
+# First variant of the design
+proto_to_dslx(
+    name = "parameters_add",
+    proto_schema_path = ":param.proto",
+    proto_name = "xls.DesignVariants",
+    textproto = ":add.textproto",
+    variable_name = "parameters",
+)
+
+# Second variant of the design
+proto_to_dslx(
+    name = "parameters_sub",
+    proto_schema_path = ":param.proto",
+    proto_name = "xls.DesignVariants",
+    textproto = ":sub.textproto",
+    variable_name = "parameters",
+)
+
+# Define config settings for selecting design variant
+string_flag(
+    name = "design_variant",
+    build_setting_default = "add",
+)
+
+config_setting(
+    name = "design_add",
+    flag_values = {
+        ":design_variant": "add",
+    },
+)
+
+config_setting(
+    name = "design_sub",
+    flag_values = {
+        ":design_variant": "sub",
+    },
+)
+
+# Move selected variant to location matching import statement in DSLX design
+genrule(
```

Comment:
curious why we need the intermediate genrules? If we emit a `DslxInfo` from the `proto_to_dslx` rule, we should be able to plug it directly into a `dslx_library`.

Reply:
This is only for the purpose of allowing easier builds for two different variants of the same design. The problem is that in top-level DSLX file ([here](https://github.com/google/xls/pull/1183/files#diff-c1e7fa11e9f9c66b0c4be19423d8c4937677ed1cd56a171712cdc431a52fa612R1)) we need to provide import path to our DSLX file with parameters. In this example I wanted to provide two different variants so that it would be visible that the implemented design is indeed changing depending on which textproto file was used for the build.
The import line must point to a file with specific name. Bazel does not allow specifying two rules (`proto_to_dslx`) that would generate the same file from two different textproto configurations.
My workaround for this is to generate files with different names and then pick one based on custom build flag and `select()` and then copy the file to generic DSLX file used in the import statement in top-level DSLX

Reply:
rather than using select, I think as a minimal example it would be simpler to have two `xls_dlsx_library` variant "top" that depends on the `proto_to_dslx` targets (to show the direct connection between the two rules)

Reply:
I was thinking about that solution before opening the PR. The thing is that I thought that having 2 top level DSLX files defeats the purpose of having parameterized design in the first place because of doubling the code. It was important for me to show that you can use a single DSLX top level design and achieve different final results depending on which textproto was used to get design parameters.
Having said that, I don't have a strong opinion here. For the sake of simplicity we can switch to example with 2 top-level designs and without select statements.

Reply:
>  I think as a minimal example it would be simpler to have two xls_dlsx_library variant "top" that depends on the proto_to_dslx targets

I've reworked the example to have 2 separate top-level designs. Please take a look @proppy